matrix_client/crypto/olm_device.py

```diff
@@ -0,0 +1,99 @@
+import logging
+
+import olm
+from canonicaljson import encode_canonical_json
+
+from matrix_client.checks import check_user_id
+
+logger = logging.getLogger(__name__)
+
+
+class OlmDevice(object):
+    """Manages the Olm cryptographic functions.
+
+    Has a unique Olm account which holds identity keys.
+
+    Args:
+        api (MatrixHttpApi): The api object used to make requests.
+        user_id (str): Matrix user ID. Must match the one used when logging in.
+        device_id (str): Must match the one used when logging in.
+    """
+
+    def __init__(self, api, user_id, device_id):
+        self.api = api
+        check_user_id(user_id)
+        self.user_id = user_id
+        self.device_id = device_id
+        self.olm_account = olm.Account()
+        self.olm_account.create()
+        logger.info('Initialised Olm Device.')
+
+    def sign_json(self, json):
+        """Signs a JSON object.
+
+        As specified, this is done by encoding the JSON object without ``signatures`` or
```

Comment:
We should probably explicitly state somewhere in the docstring whether this can be used as a mutating function or if one must take the return value.

Reply:
done